### PR TITLE
IDE Themes

### PR DESCRIPTION
**Pull request by dabbott**
MEGA PR

- Introduces themes using https://github.com/dabbott/react-styles-provider
- Update the default editor theme
- Add a dark UI theme (light is still default)
- Clean up TabbedEditor

### CODE REVIEW

File: web/src/scripts/containers/TabbedEditor.jsx [hunk 1]
```diff
@@ -126,20 +222,61 @@ class TabbedEditor extends Component {
...
+  }
+
+  saveScrollHeight = (scrollHeight) => this.props.dispatch(setConsoleScrollHeight(scrollHeight))
+
+  // Delay allows key events to finish first?
+  // TODO: move search menu to top level and take care of this on that refactor
+  onRequestCloseSearchMenu = () => {
+    setTimeout(() => {
+      this.refs.editor.getDecoratedComponentInstance().focus()
+    }, 200)
```
Comment (gabergg) [inline web/src/scripts/containers/TabbedEditor.jsx]: can probably delay 0 on things like this?

Comment (dabbott) [inline web/src/scripts/containers/TabbedEditor.jsx]: This is some ancient code. I don't think delaying 0 works. CodeMirror internally delays 10 or 50 something to batch events.

File: web/src/scripts/components/menu/Menu.jsx [hunk 2]
```diff
@@ -195,17 +315,25 @@ class Menu extends Component {
...
-      <MenuInner children={this.props.children}
-        className={this.props.className}
-        style={this.props.style}
-        caret={this.props.caret}
-        caretPosition={this.state.caretPosition}
-        onResize={this._onInnerContentResize.bind(this)}
-        onClick={this._dismissOnClickChildren.bind(this)}
-        hideMenu={this._dismissOnClickChildren.bind(this)}
-        transitionName={this.props.transitionName} />
+      <StylesProvider theme={this.context.theme}>
```
Comment (gabergg) [inline web/src/scripts/components/menu/Menu.jsx]: can you briefly explain this to me?

Comment (dabbott) [inline web/src/scripts/components/menu/Menu.jsx]: sure

File: web/src/scripts/components/filetree/NodeCaret.js [hunk 3]
```diff
@@ -23,15 +30,17 @@ export default class extends Component {
     return shallowCompare(this, nextProps, nextState)
   }
 
+  onMouseDown = () => (e) => e.stopPropagation()
```
Comment (gabergg) [inline web/src/scripts/components/filetree/NodeCaret.js]: (e) => e.stopPropagation()?

Comment (dabbott) [inline web/src/scripts/components/filetree/NodeCaret.js]: lol, yes

**Pull request merged**